Scripts/RemediationScripts/Remediate-EnableHttpsForStorageAccount.ps1

```diff
@@ -0,0 +1,723 @@
+            Break
+        }
+
+        if($resourceResolver.messageToPrint -ne $null)
+        {
+            $resourceSummary += "Excluded resource/resource group summary:`n " 
+            $resourceSummary += $resourceResolver.messageToPrint
+        }   
+    }
+    
+    Write-Host $resourceSummary
+    Write-Host "Total Storage Account(s) excluded  from remediation:" [$($totalStorageAccount - ($resourceContext | Measure-Object).Count)] -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host "Total Storage Account(s) for remediation: [$(($resourceContext | Measure-Object).Count)]" -ForegroundColor $([Constants]::MessageType.Update)
+    write-host "$([Constants]::DoubleDashLine)"
+
+    try
+    {
+        #Storage Account with enabled HTTPs only
+        $stgWithEnableHTTPs = @()
+        #Storage Account with disabled HTTPs only
+        $stgWithDisableHTTPs = @()
+        #Storage Account skipped from remediation
+        $skippedStorageAccountFromRemediation = @()
+
+        Write-Host "[step 4 of 6] : Getting the Count of Storage Account in a subscription which are having 'EnableHttpsTrafficOnly' value set to false."
+        $resourceContext | ForEach-Object {
+            if ($_.EnableHttpsTrafficOnly)
+            {
+                $stgWithEnableHTTPs += $_ 
+            }
+        
+            else
+            { 
+                 $stgWithDisableHTTPs += $_|Select-Object  @{N='ResourceId';E={$_.Id}},
+                                                           @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                           @{N='StorageAccountName';E={$_.StorageAccountName}},
+                                                           @{N='EnableHttpsTrafficOnly';E={$_.EnableHttpsTrafficOnly}}
+                                                                              
+            }
+        }                    
+                              
+    
+        $totalstgWithEnableHTTPs = ($stgWithEnableHTTPs | Measure-Object).Count
+        $totalstgWithdisableHTTPs = ($stgWithDisableHTTPs | Measure-Object).Count
+    
+        Write-Host "Storage Account(s) with enabled 'EnablettpsTrafficOnly': [$($totalstgWithEnableHTTPs)]" -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host "Storage Account(s) with disabled 'EnableHttpsTrafficOnly'': [$($totalstgWithdisableHTTPs)]" -ForegroundColor $([Constants]::MessageType.Update)
+
+        #Start remediation Storage Account(s) with 'EnableHttpsTrafficOnly' enabled.
+            if ($totalstgWithDisableHTTPs -gt 0)
+            {
+                # Creating the log file
+                if ($DryRun)
+                {  
+                    write-host " $([Constants]::SingleDashLine)"
+                    Write-Host "Exporting configurations of Storage Account(s)  for which  the value of 'EnableHTTPsTrafficonly' is false.You may want to use this CSV as a pre-check before actual remediation." 
```

Comment:
To reiterate, a more appropriate way could be - "Exporting configurations of Storage Account(s) that don't have secure transfer enabled..."

Reply:
Done